### PR TITLE
[Modules 5519] Add port parameter in class mod::jk

### PR DESCRIPTION
**Pull request by EmersonPrado**
New parameters allow changing binding settings for `mod_jk`. This way, one can use Apache with `mod_jk` and a reverse proxy / cache receiving requests and forwarding to Apache in a different port, for example.

### CODE REVIEW

**Comment by EmersonPrado** [comment]
Finally, a working spec test with probably enough contexts

File: README.md [hunk 1]
```diff
@@ -2165,7 +2166,31 @@ class { '::apache::mod::jk':
 
 **Parameters within `apache::mod::jk`**:
 
-The best source for understanding the `mod_jk` parameters is the [official documentation](https://tomcat.apache.org/connectors-doc/reference/apache.html), except for \*file_content:
+The best source for understanding the `mod_jk` parameters is the [official documentation](https://tomcat.apache.org/connectors-doc/reference/apache.html), except for:
```
Comment (eputnam) [inline README.md]: yes! we like linking to official docs!

File: manifests/mod/jk.pp [hunk 2]
```diff
@@ -9,6 +9,11 @@
 # https://tomcat.apache.org/connectors-doc/reference/apache.html
 #
 class apache::mod::jk (
+  # Binding to mod_jk
+  Optional[String] $ip         = undef,
+  Pattern[/^\d+$/] $port       = '80',
+  Boolean          $add_listen = true,
+  # Conf file content
   $workers_file          = undef,
```
Comment (eputnam) [inline manifests/mod/jk.pp]: please align the rest of these `=`

File: manifests/mod/jk.pp [hunk 3]
```diff
@@ -9,6 +9,11 @@
 # https://tomcat.apache.org/connectors-doc/reference/apache.html
 #
 class apache::mod::jk (
+  # Binding to mod_jk
+  Optional[String] $ip         = undef,
+  Pattern[/^\d+$/] $port       = '80',
```
Comment (eputnam) [inline manifests/mod/jk.pp]: is there a particular reason you're not using the `Integer` type?

Comment (EmersonPrado) [inline manifests/mod/jk.pp]: The variable `$port` is used as a string - interpolated in another string in a `ensure_resource` [declaration](https://github.com/puppetlabs/puppetlabs-apache/pull/1679/files#diff-adb49d5dc0194dcc9856f3a221521c23R76)

File: spec/classes/mod/jk_spec.rb [hunk 4]
```diff
@@ -12,13 +12,19 @@
     it { is_expected.to contain_file('jk.conf').that_notifies('Class[apache::service]') }
   end
 
-  context "RHEL 6 with only required facts and no parameters" do
+  default_ip = '192.168.1.1'
```
Comment (EmersonPrado) [inline spec/classes/mod/jk_spec.rb]: I had a problem in #1671 regarding this change. I'll wait 'till we solve that then replicate the solution here.

File: manifests/mod/jk.pp (hunk 2 above)
Comment (EmersonPrado) [inline manifests/mod/jk.pp]: Done!

**Comment by eputnam** [comment]
@EmersonPrado could you please rebase this on master? basically just adding this work to your previous PR for paths.

**Comment by EmersonPrado** [comment]
I rebased and pushed again, but it didn't come out right:
```
 ! [rejected]        MODULES-5519 -> MODULES-5519 (non-fast-forward)
error: failed to push some refs to 'https://github.com/EmersonPrado/puppetlabs-apache.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
```

And now the PR has an unsolved conflict. I did solve all conflicts locally while rebasing, so I believe to have pushed a clean tree. Do you have a hint? It's the first time I rebase something I have already pushed (in fact, I always thought I shouldn't), so I don't really know how to solve this.

**Comment by EmersonPrado** [comment]
I think I could solve all conflicts manually from the Web interface. Are we done?

**Comment by ekohl** [comment]
@EmersonPrado you have to force push (```git push -f```) after a rebase. Since you've rewritten the history, it's no longer fast-forward. Doing non-fast-forward pushes on shared branches (master, release branches) is bad, hence the warning, but on PR branches it's fine.

**Pull request merged**